src/pydicom/dataset.py

```diff
@@ -2786,6 +2790,58 @@ def __setitem__(self, key: "slice | TagType", elem: _DatasetValue) -> None:
             #   dataset
             self._set_pixel_representation(cast(DataElement, elem))
 
+    def set_pixel_data(
+        self,
+        arr: "numpy.ndarray",
+        photometric_interpretation: str,
+        bits_stored: int,
+    ) -> None:
+        """Use an :class:`~numpy.ndarray` to set the *Pixel Data* and related
+        Image Pixel module elements.
+
+        .. versionadded:: 3.0
+
+        The following :dcm:`Image Pixel<part03/sect_C.7.6.3.3.html#table_C.7-11c>`
+        module elements values will be added, updated or removed as necessary:
+
+        * (0028,0002) *Samples per Pixel* using a value corresponding to
+          `photometric_interpretation`.
+        * (0028,0104) *Photometric Interpretation* from `photometric_interpretation`.
+        * (0028,0006) *Planar Configuration* will be added and set to ``0`` if
+          *Samples per Pixel* is > 1, otherwise it will be removed.
+        * (0028,0008) *Number of Frames* from the array :attr:`~numpy.ndarray.shape`,
+          however it will be removed if `arr` only contains a single frame.
+        * (0028,0010) *Rows* and (0028,0011) *Columns* from the array
+          :attr:`~numpy.ndarray.shape`.
+        * (0028,0100) *Bits Allocated* from the array :class:`~numpy.dtype`.
+        * (0028,0101) *Bits Stored* and (0028,0102) *High Bit* from `bits_stored`.
+        * (0028,0103) *Pixel Representation* from the array :class:`~numpy.dtype`.
+
+        In addition, the *Transfer Syntax UID* will be set to *Explicit VR Little
+        Endian* if it doesn't already exist or uses a compressed (encapsulated)
+        transfer syntax.
```

Comment:
What about SOPInstanceUID?

Reply:
Good point, I also need to either disallow Big Endian or account for it.

Reply:
Done